tools/ot/src/main/java/dk/alexandra/fresco/tools/ot/base/NaorPinkasOt.java

```diff
@@ -0,0 +1,323 @@
+  }
+
+  @Override
+  public StrictBitVector receive(Boolean choiceBit) {
+    byte[] seed = receiveByteOt(choiceBit);
+    byte[] encryptedZeroMessage = network.receive(otherId);
+    byte[] encryptedOneMessage = network.receive(otherId);
+    return recoverTrueMessage(encryptedZeroMessage, encryptedOneMessage, seed,
+        choiceBit);
+  }
+
+  /**
+   * Make a one-time padding of a {@code message} using a PRG seeded on
+   * {@code seed}.
+   *
+   * @param message
+   *          The message to one-time pad
+   *
+   * @param maxSize
+   *          The size of the resultant ciphertext. MUST be at least the length
+   *          of the message itself. If longer, it will be padded with zeros
+   * @param seed
+   *          The seed to use for a PRG
+   * @return The one-time padded message
+   */
+  private byte[] padMessage(byte[] message, int maxSize,
+      byte[] seed) {
+    byte[] maxLengthMessage = Arrays.copyOf(message, maxSize);
+    byte[] encryptedMessage = new byte[maxSize];
+    Drbg prg = new AesCtrDrbg(seed);
+    prg.nextBytes(encryptedMessage);
+    ByteArrayHelper.xor(encryptedMessage, maxLengthMessage);
+    return encryptedMessage;
+  }
+
+  /**
+   * Decrypt a one-time padded {@code paddedMessage} using a PRG seeded on
+   * {@code seed}.
+   *
+   * @param paddedMessage
+   *          The message to one-time pad
+   * @param seed
+   *          The seed to use for a PRG
+   * @return The plain message
+   */
+  private byte[] unpadMessage(byte[] paddedMessage, byte[] seed) {
+    byte[] message = new byte[paddedMessage.length];
+    Drbg prg = new AesCtrDrbg(seed);
+    prg.nextBytes(message);
+    ByteArrayHelper.xor(message, paddedMessage);
+    return message;
+  }
+
+  /**
+   * Receive one-time padded OT messages and remove the pad of the one of the
+   * messages chosen in the OT.
+   *
+   * @param encryptedZeroMessage
+   *          The one-time padded zero-message
+   * @param encryptedOneMessage
+   *          the one-time padded one-message
+   * @param seed
+   *          The seed used for padding of one of the messages
+   * @param choiceBit
+   *          A bit indicating which message the seed matches. False implies
+   *          message zero and true message one.
+   * @return The unpadded message as a StrictBitVector
+   */
+  private StrictBitVector recoverTrueMessage(byte[] encryptedZeroMessage,
+      byte[] encryptedOneMessage, byte[] seed, boolean choiceBit) {
+    if (encryptedZeroMessage.length != encryptedOneMessage.length) {
+      throw new MaliciousException(
+          "The length of the two choice messages is not equal");
+    }
+    byte[] unpaddedMessage;
+    if (choiceBit == false) {
+      unpaddedMessage = unpadMessage(encryptedZeroMessage, seed);
+    } else {
+      unpaddedMessage = unpadMessage(encryptedOneMessage, seed);
+    }
+    return new StrictBitVector(unpaddedMessage);
+  }
+
+  /**
+   * Completes the sender's part of the Naor-Pinkas OT.
```

Comment:
This does not really complete the senders part. He does some more stuff in the `send()` method.

Reply:
That is true. But what he does in send() is not really Naor-Pinkas, but a standard padding approach to use the random messages from Naor-Pinkas as a regular 1-out-of-2 OT. 
I have added some more code comments to clarify in commit 21e9cb27